src/counter.cc

```diff
@@ -0,0 +1,432 @@
+// Copyright 2015 Google Inc. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "benchmark/benchmark.h"
+#include "check.h"
+#include "internal_macros.h"
+
+#include <cstring>
+
+namespace benchmark {
+
+Counter::Counter()
+    : name_(nullptr),
```

Comment:
i think all the supported compilers allow delegating constructors now so feel free to use them.